code/modules/mob/living/simple_animal/hostile/space_dragon.dm

```diff
@@ -0,0 +1,727 @@
+ */
+/mob/living/simple_animal/hostile/space_dragon/proc/rift_empower(is_permanent)
+	fully_heal()
+	add_filter("anger_glow", 3, list("type" = "outline", "color" = "#ff330030", "size" = 5))
+	add_movespeed_modifier(MOVESPEED_ID_DRAGON_RAGE, multiplicative_slowdown = -0.5)
+	addtimer(CALLBACK(src, .proc/rift_depower), 30 SECONDS)
+
+/**
+ * Gives Space Dragon their the rift speed buff permanantly.
+ *
+ * Gives Space Dragon the enraged speed buff from charging rifts permanantly.
+ * Only happens in circumstances where Space Dragon completes their objective.
+ */
+/mob/living/simple_animal/hostile/space_dragon/proc/permanant_empower()
+	fully_heal()
+	add_filter("anger_glow", 3, list("type" = "outline", "color" = "#ff330030", "size" = 5))
+	add_movespeed_modifier(MOVESPEED_ID_DRAGON_RAGE, multiplicative_slowdown = -0.5)
+
+/**
+ * Removes Space Dragon's rift speed buff.
+ *
+ * Removes Space Dragon's speed buff from charging a rift.  This is only called
+ * in rift_empower, which uses a timer to call this after 30 seconds.  Also
+ * removes the red glow from Space Dragon which is synonymous with the speed buff.
+ */
+/mob/living/simple_animal/hostile/space_dragon/proc/rift_depower()
+	remove_filter("anger_glow")
+	remove_movespeed_modifier(MOVESPEED_ID_DRAGON_RAGE)
+
+/**
+ * Destroys all of Space Dragon's current rifts.
+ *
+ * QDeletes all the current rifts after removing their references to other objects.
+ * Currently, the only reference they have is to the Dragon which created them, so we clear that before deleting them.
+ * Currently used when Space Dragon dies or one of his rifts is destroyed.
+ */
+/mob/living/simple_animal/hostile/space_dragon/proc/destroy_rifts()
+	rifts_charged = 0
+	add_movespeed_modifier(MOVESPEED_ID_DRAGON_DEPRESSION, multiplicative_slowdown = 5)
+	riftTimer = -1
+	tiredness_mult = 5
+	playsound(src, 'sound/vehicles/rocketlaunch.ogg', 100, TRUE)
+	for(var/obj/structure/carp_rift/rift in rift_list)
+		rift.dragon = null
+		rift_list -= rift
+		if(!QDELETED(rift))
+			QDEL_NULL(rift)
+
+/**
+ * Handles wing gust from the windup all the way to the endlag at the end.
+ *
+ * Handles the wing gust attack from start to finish, based on the timer.
+ * When intially triggered, starts at 0.  Until the timer reaches 10, increase Space Dragon's y position by 2 and call back to the function in 1.5 deciseconds.
+ * When the timer is at 10, trigger the attack.  Change Space Dragon's sprite. reset his y position, and push all living creatures back in a 3 tile radius and stun them for 5 seconds.
+ * Stay in the ending state for how much our tiredness dictates and add to our tiredness.
+ * Arguments:
+ * * timer - The timer used for the windup.
+ */
+/mob/living/simple_animal/hostile/space_dragon/proc/useGust(timer)
+	if(timer != 10)
+		pixel_y = pixel_y + 2;
+		addtimer(CALLBACK(src, .proc/useGust, timer + 1), 1.5)
+		return
```

Comment:
Use animate instead of calling itself over and over again with a timer.
Also 1.5 seconds is a long time to be frozen for in order to use this ability, you will probably have lost half your health by then, or at least everyone will be out of the way.

Reply:
Changed to 1 second, plus there was no cooldown so I added one for 3 seconds.